packages/deployer/src/compiler.ts

```diff
@@ -193,20 +225,24 @@ export class Compiler {
                 });
             }
         }
-        const contractName = path.basename(fileName, constants.SOLIDITY_FILE_EXTENSION);
-        const contractIdentifier = `${fileName}:${contractName}`;
+        const contractName = path.basename(sourceFileId, constants.SOLIDITY_FILE_EXTENSION);
+        const contractIdentifier = `${sourceFileId}:${contractName}`;
         if (_.isUndefined(compiled.contracts[contractIdentifier])) {
             throw new Error(
-                `Contract ${contractName} not found in ${fileName}. Please make sure your contract has the same name as it's file name`,
+                `Contract ${contractName} not found in ${sourceFileId}. Please make sure your contract has the same name as it's file name`,
             );
         }
-        const abi: ContractAbi = JSON.parse(compiled.contracts[contractIdentifier].interface);
+        const abiMaybeWithOverloadedFunctions: ContractAbi = JSON.parse(
```

Comment:
I'm wondering if this is something that should be fixed at the `abi-gen` level, rather than the ABI level. This change makes our compiler output non-standard ABIs. Thoughts @LogvinovLeon @Recmo @fabioberger ?

Reply:
Agreed.